introspection-engine/introspection-engine-tests/tests/re_introspection/relation_mode/postgres.rs

```diff
@@ -0,0 +1,376 @@
+use indoc::indoc;
+use introspection_engine_tests::test_api::*;
+
+// referentialIntegrity = "prisma" preserves the relation policy ("prisma") as well as @relations.
+#[test_connector(tags(Postgres))]
+async fn referential_integrity_prisma(api: &TestApi) -> TestResult {
+    let init = formatdoc! {r#"
+        CREATE TABLE "Foo" (
+            "id" INTEGER NOT NULL,
+            "bar_id" INTEGER NOT NULL,
+        
+            CONSTRAINT "Foo_pkey" PRIMARY KEY ("id")
+        );
+        
+        CREATE TABLE "Bar" (
+            "id" INTEGER NOT NULL,
+        
+            CONSTRAINT "Bar_pkey" PRIMARY KEY ("id")
+        );
+        
+        CREATE UNIQUE INDEX "Foo_bar_id_key" ON "Foo"("bar_id");
+    "#};
+
+    api.raw_cmd(&init).await;
+
+    let input = indoc! {r#"
+        generator client {
+            provider        = "prisma-client-js"
+            previewFeatures = ["referentialIntegrity"]
+        }
+
+        datasource db {
+            provider             = "postgres"
+            url                  = env("TEST_DATABASE_URL")
+            referentialIntegrity = "prisma"
+        }
+
+        model Foo {
+            id     Int @id
+            bar    Bar @relation(fields: [bar_id], references: [id])
+            bar_id Int @unique
+        }
+
+        model Bar {
+            id  Int  @id
+            foo Foo?
+        }
+    "#};
+
+    let expected = expect![[r#"
+        generator client {
+          provider        = "prisma-client-js"
+          previewFeatures = ["referentialIntegrity"]
+        }
+
+        datasource db {
+          provider             = "postgres"
+          url                  = env("TEST_DATABASE_URL")
+          referentialIntegrity = "prisma"
+        }
+
+        model Foo {
+          id     Int @id
+          bar    Bar @relation(fields: [bar_id], references: [id])
+          bar_id Int @unique
+        }
+
+        model Bar {
+          id  Int  @id
+          foo Foo?
+        }
+    "#]];
+
+    let result = api.re_introspect_config(input).await?;
+    expected.assert_eq(&result);
+
+    Ok(())
+}
+
+// referentialIntegrity = "prisma" preserves the relation policy ("foreignKeys") as well as @relations, which are moved to the bottom.
```

Comment:
```suggestion
// referentialIntegrity = "foreignKeys" preserves the relation policy ("foreignKeys") as well as @relations, which are moved to the bottom.
```

Reply:
Will commit with a `[skip ci]` message once the CI is green.